int16 cmd0
int16 cmd1